File: easy/1207. Unique Number of Occurrences/1207. Unique Number of Occurrences.go
package main

func uniqueOccurrences(arr []int) bool {
	items := make(map[int]int)
	for _, n := range arr {
		items[n] += 1
	}

	occurs := make(map[int]bool)

	for k := range items {
		if occurs[items[k]] {
			return false
		}
		occurs[items[k]] = true
	}

	return true
}

File: easy/1207. Unique Number of Occurrences/1207. Unique Number of Occurrences.py
from collections import defaultdict
from typing import List


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        items = defaultdict(int)

        for num in arr:
            items[num] += 1

        occurs = set()
        for _, o in items.items():
            if o in occurs:
                return False
            occurs.add(o)

        return True
